src/SINDYPI_fit.jl

```diff
@@ -0,0 +1,457 @@
+##FIT FUNCTION
+
+function fit!(X::AbstractArray, Y::AbstractArray, maxiter, lambda, Polyorder, Trigorder)
+    using LinearAlgebra
+    # Compute awesome stuff here
+    s = []
+    for i = 1:1:size(X,2)
+        push!(s, string("x", string(i)))
+    end
+
+number_states_guess = size(X, 2)
+
+Store_Result = Array{Any}(undef, number_states_guess)
+Index_nonzero = Array{Any}(undef, number_states_guess)
+Xi = Array{Any}(undef, number_states_guess)
+
+
+ for w = 1:number_states_guess

```

Comment:
Here you are deriving a basis ( I suspect based on the code in the sindy matlab implementations). 

In DataDrivenDiffEq we provide a `Basis` struct for this. So while this here is nice for automatic construction of a subset of candidates ( which should be also possible in the package, but ain't nobody got time for this :) ), we build around the structured, well known input of a basis.

Reply:
A basis also generates optimized code via ModelingToolkit and bridges functionality for all algorithms.